openedx/openedx-platform
# [BD-04] Convert PollModule to PollBlock and CustomTagModule to CustomTagBlock

## Description

Converts the Poll XModule and CustomTagXModule into XBlocks.

Part of [XModule to XBlock Conversion work](https://openedx.atlassian.net/wiki/spaces/AC/pages/1472790755/XModule+to+XBlock+Conversion). 

- Which edX user roles will this change impact? "Developer"

## Testing Instructions

* Checkout the master branch.
* Create a course and import this archive: [course-xmodules.tar.gz](https://github.com/edx/edx-platform/files/6092077/course-xmodules.tar.gz)
* The course has an example Poll XBlock and an example CustomTag XBlock. These XBlocks cannot be created in Studio so we need to create them in XML and import.
* View and interact with both of them in Studio and LMS.
* Checkout this branch and run `pip install -e common/lib/xmodule/` and `NO_PREREQ_INSTALL=1 paver update_assets` and then `make lms-restart` + `make studio-restart`.
* View and interact with both of them in Studio and LMS. They should look and behave the same (see screenshots below).
* Export the course and check that the OLX of Poll and CustomTag is the same as the OLX in the original archive.
* Import the exported course and check that the two XBlocks are still a part of the course.

## Screenshots

All the views should look and work the same before and after the conversion.

**CustomTag author view**

![image](https://user-images.githubusercontent.com/5305552/110149541-ffff8380-7dff-11eb-9c25-52e61e55e2d3.png)

**CustomTag studio view**

![image](https://user-images.githubusercontent.com/5305552/110149679-26bdba00-7e00-11eb-8bfb-64a79c501443.png)

**CustomTag student view**

![image](https://user-images.githubusercontent.com/5305552/110149811-4c4ac380-7e00-11eb-8b5a-55346dc78d57.png)

**Poll author view**

![image](https://user-images.githubusercontent.com/5305552/110149864-5c62a300-7e00-11eb-93b2-eb73da927bb4.png)

**Poll studio view** (It does not exist so we get this error)

![image](https://user-images.githubusercontent.com/5305552/110149931-72706380-7e00-11eb-84ec-f6982bef7060.png)

**Poll student view**

![image](https://user-images.githubusercontent.com/5305552/110150025-8e740500-7e00-11eb-801c-935459f78aee.png)
